BUILD.bazel

```diff
@@ -0,0 +1,337 @@
+    visibility = ["//visibility:public"],
+    copts = [
+        "-Iexternal/qt/QtCore",
+        "-Iexternal/qt/QtGui",
+        "-Iexternal/qt/QtQuick",
+        "-Iexternal/qt/QtQuickControls2",
+        "-Iexternal/qt/QtWidgets",
+        "-Iexternal/qt/QtQml",
+    ],
+    deps = [
+        "//ign_common/events:events",
+        "//ign_common",
+        "//ign_math",
+        "//ign_msgs",
+        "//ign_bazel:utilities",
+        "//ign_plugin/loader:loader",
+        "//ign_plugin/core:ign_plugin",
+        "@qt//:qt_core",
+        "@qt//:qt_network",
+        "@qt//:qt_widgets",
+        "@qt//:qt_quick_control",
+        "@qt//:qt_quick",
+        "@qt//:qt_qml",
+        "@qt//:qt_gui",
+        "@qt//:qt_opengl",
+    ],
+    defines = select({
+       "//ign_transport:custom": ["HAVE_IFADDRS"],
+       "//conditions:default": []
+    }),
+)
+
+'''
+qt_cc_library(
+    name = "ign_gui",
+    srcs = sources + public_headers,
+    hdrs = public_headers,
+    includes = ["include"],
+    copts = [
+        "-Iexternal/qt/QtCore",
+        "-Iexternal/qt/QtGui",
+        "-Iexternal/qt/QtQuick",
+        "-Iexternal/qt/QtQuickControls2",
+        "-Iexternal/qt/QtWidgets",
+        "-Iexternal/qt/QtQml",
+    ],
+    deps = [
+        ":main-resources",
+        "//ign_common/events",
+        "//ign_common",
+        "//ign_math",
+        "//ign_msgs",
+        "//ign_bazel:utilities",
+        "//ign_plugin/loader",
+        "//ign_plugin/core:ign_plugin",
+        "@qt//:qt_core",
+        "@qt//:qt_network",
+        "@qt//:qt_widgets",
+        "@qt//:qt_quick_control",
+        "@qt//:qt_quick",
+        "@qt//:qt_qml",
+        "@qt//:qt_gui",
+        "@qt//:qt_opengl",
+    ],
+)
+
+
+
+qt_cc_library(
+    name = "ign_gui",
+
+)
+
+
+cc_binary(
+    name = "libignition-gui4.so",
+    srcs = sources + public_headers,
+    includes = ["include"],
+    linkopts = [
+        "-Wl,-soname,libignition-gui4.so",
+        "-ltinyxml2",
+    ],
+    copts = [
+        "-Iexternal/qt/QtCore",
+        "-Iexternal/qt/QtGui",
+        "-Iexternal/qt/QtQuick",
+        "-Iexternal/qt/QtQuickControls2",
+        "-Iexternal/qt/QtWidgets",
+        "-Iexternal/qt/QtQml",
+    ],
+    linkshared = True,
+    visibility = [],
+    deps = [
+        "//ign_common/events:events",
+        "//ign_common",
+        "//ign_math",
+        "//ign_msgs",
+        "//ign_bazel:utilities",
+        "//ign_plugin/loader:loader",
+        "//ign_plugin/core:ign_plugin",
+        "@qt//:qt_core",
+        "@qt//:qt_network",
+        "@qt//:qt_widgets",
+        "@qt//:qt_quick_control",
+        "@qt//:qt_quick",
+        "@qt//:qt_qml",
+        "@qt//:qt_gui",
+        "@qt//:qt_opengl",
+    ],
+)
+cc_library(
+    name = "ign_gui",
+    srcs = ["libignition-gui4.so"],
+    hdrs = public_headers,
+    includes = ["include"],
+    visibility = ["//visibility:public"],
+    linkopts = [
+        "-Wl,-soname,libignition4-gui.so",
+        "-ltinyxml2",
```

Comment:
I believe that you should be able to use `@tinyxml2` in the deps, rather than linkopts

Reply:
That doesn't quite seem to be working for me

Reply:
Okay, we can skip this for now.